### PR TITLE
#164326073 Configure reset password confirm url and write tests.

### PR DESCRIPTION
**Pull request by katherine95**
#### What does this PR do?
Has tests for password reset endpoint
#### Description of Task to be completed?
Configure `/api/auth/reset_password_confirm/` url and write tests for the endpoint.
#### How should this be manually tested?
1. Clone this repo.
   `git clone https://github.com/jkamz/ireporter.git`
2. Fetch all origin to get all branches
   `git fetch origin`
3. Checkout to `ft-reset-password-email-164326073` branch.
   `git checkout ft-reset-password-email-164326073`
4. Activate `virtualenv`
    `$ virtualenv -p python .venv`
    `$ source .venv/bin/activate`
5. Install requirements
   `$ pip install -r requirements.txt`
6. Start server
   `$ python api/manage.py runserver`
7. Test the endpoint on postman in the following steps:
      - `/api/auth/signup/`
      - `/api/auth/activate/`
      - `/api/auth/reset_password/`
      -  `/api/auth/reset_password_confirm?uid=string_uid&token=generated_token` url)
8. Run tests:
   `$ python api/manage.py test`
#### What are the relevant pivotal tracker stories?
[#164326073](https://www.pivotaltracker.com/story/show/164326073)
#### Screenshot
image

### CODE REVIEW

**Comment by Siffersari** [comment]
Good job

**Comment by meshack-mbuvi** [comment]
Instead of having to pass the generated token and uid generated in the /api/auth/reset_password_confirm?uid=string_uid&token=generated_token and passing them in the body, you can just leave the url the way it is, pass the new password only, then in your view, extract the query paramaters. This can help you: https://www.django-rest-framework.org/tutorial/2-requests-and-responses/

**Comment by Benkimeric** [comment]
Nice job Catherine

File: api/auth/views.py [hunk 1]
```diff
@@ -118,6 +118,7 @@ class LoginView(utils.ActionViewMixin, generics.GenericAPIView):
     def _action(self, serializer):
         token = utils.login_user(self.request, serializer.user)
         token_serializer_class = serializers.TokenSerializer
+        print(token_serializer_class.data)
```
Comment (meshack-mbuvi) [inline api/auth/views.py]: The print statement is not required. Please remove it.

File: api/test/data/test_reset_password.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            activate_data = self.activation_data
+
+        # activate the account
+        self.client.post(
+            reverse('user_activate'),
+            self.activation_data,
+            format="json"
+        )
+
+        #reset password
```
Comment (meshack-mbuvi) [inline api/test/data/test_reset_password.py]: This comment is not needed; ideally, your code should be self-descriptive.

File: api/test/data/test_reset_password.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        self.client.post(
+            reverse('reset_password'),
+            self.user_data['email'],
+            format="json"
+        )
+
+        #reset confirm password
```
Comment (meshack-mbuvi) [inline api/test/data/test_reset_password.py]: same here

File: api/test/data/test_reset_password.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+        self.assertEqual(self.response.status_code,
+                         status.HTTP_200_OK)
+
+    def test_user_cannot_reset_password_with_account_not_activated(self):
+        """
+        Test method to ensure users cannot reset password with unverified
+        email address
+        """
+        # sign-up user
```
Comment (meshack-mbuvi) [inline api/test/data/test_reset_password.py]: same here

File: api/test/data/test_reset_password.py (hunk 3 above)
Comment (katherine95) [reply api/test/data/test_reset_password.py]: resolved

File: api/test/data/test_reset_password.py (hunk 4 above)
Comment (katherine95) [reply api/test/data/test_reset_password.py]: resolved

**Pull request merged**